### PR TITLE
Add basic public API Gemini support

### PR DESCRIPTION
**Pull request by github-ajt**
Unit tests added for Gemini public API calls.

Missing Gemini feed and private API calls (in a later PR).

### CODE REVIEW

File: src/exchanges/gemini/GeminiExchangeAPI.ts [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+        throw new Error('Method not implemented.');
+    }
+
+    private publicRequest(command: string): Promise<any> {
+        const url = `${this.apiURL}${command}`;
+        return request.get(url)
+            .accept('application/json')
+            .then((response) => {
+                if (response.status !== 200) {
+                    throw new Error('publicRequest for ' + command + ' was not successful');
```
Comment (Ameobea) [inline src/exchanges/gemini/GeminiExchangeAPI.ts]: It may be a good idea to include the status code in the error message.

Comment (github-ajt) [reply src/exchanges/gemini/GeminiExchangeAPI.ts]: Done.

**Comment by CjS77** [comment]
Since we have ccxt as a dependency now, it's fairly easy to write a wrapper for any exchange -- see the refactoring for Poloniex.

I'd prefer to go this route since it let's us get on with writing higher-level tools for the GTT and the ccxt maintainers can focus on keeping the API interfaces up-to-date.

Have a look and see if you can find any good reasons not to use ccxt going forward (Public and Auth calls should be working out of the box).

Note that we still need ws feed implementations for exchanges, since that's not supported by the ccxt (I suspect it's on their roadmap though).

**Comment by github-ajt** [comment]
Thanks for the update @CjS77 - closing this PR.

**Comment by kroitor** [comment]
@CjS77 hi! 

First of all, thx for using our ccxt library! 

Cool to know it's been used in this project. We're watching this repo very closely, and yes, WS pub/sub support is on our roadmap (most of WS unification is done already, so we are going to deliver soon). 

Also, we are going to implement the entire unified API for all exchanges, of course, though, we don't have as much resources as GDAX teams may have, I guess ;) Our userbase isn't very large yet, but we hope to grow quickly ;) 

We will be happy for all feedback regarding the library, for your help and for pull requests! Let us know if you have any questions about it. Thx again!

**Pull request closed without merging**